### ISSUE TITLE
analysis server dies if it can't bind to the diagnostics port

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/sdk#25394]
Filed as https://github.com/dart-atom/dartlang/issues/682 - if the analysis server can't bind to the given diagnostics port (`--port=xxx`), it fails to start up.

```
analysis-server: Unhandled exception:
SocketException: Failed to create server socket (OS Error: Address already in use, errno = 48), address = 127.0.0.1, port = 23072
#0      _NativeSocket.bind.<anonymous closure> (dart:io-patch/socket_patch.dart:494)
#1      _RootZone.runUnary (dart:async/zone.dart:1149)
#2      _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:551)
#3      _Future._propagateToListeners (dart:async/future_impl.dart:637)
#4      _Future._completeWithValue (dart:async/future_impl.dart:424)
#5      _Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:479)
#6      _microtaskLoop (dart:async/schedule_microtask.dart:41)
```

`Dart VM version: 1.14.0-dev.5.0`

**Comment by pq** [comment]
https://codereview.chromium.org/1573343003/

**Comment by pq** [comment]
Fixed w/ 6ead1c4359afc3863e42a88e9504f3d7a6f9f7e9.
